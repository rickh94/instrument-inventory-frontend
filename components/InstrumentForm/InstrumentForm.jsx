import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { lastButton } from '../../globalStyles'
import {
  makeStyles,
  FormControl,
  InputLabel,
  Button,
  Input,
  FormHelperText,
  FormGroup,
  FormControlLabel,
  Checkbox,
} from '@material-ui/core'
import { Fields } from '..'

const useStyles = makeStyles({
  lastButton,
  fileInput: {
    display: 'none',
  },
})

const InstrumentForm = ({
  onSubmit,
  setValue,
  errors,
  setErrors,
  clearForm,
  validateForm,
  instrumentNumber,
  instrumentType,
  size,
  location,
  assignedTo,
  maintenanceNotes,
  conditionNotes,
  quality,
  condition,
  rosin,
  bow,
  shoulderRestEndpinRest,
  readyToGo,
  gifted,
  onCancel,
  buttonsLeft,
  photoField,
}) => {
  const classes = useStyles()
  const [photo, setPhoto] = useState(null)

  const handleRating = name => event => {
    if (event.target.value > 5 || event.target.value < 0) {
      setValue(event.target.value)
      setErrors(name, 'Value must be between 1 and 5')
    } else if (event.target.value == 0) {
      setValue(name, '')
      setErrors(name, null)
    } else {
      setValue(name, event.target.value)
      setErrors(name, null)
    }
  }

  const handleChange = name => event => {
    setValue(name, event.target.value)
  }

  const handleCheck = name => event => {
    setValue(name, event.target.checked)
  }

  const handlePhoto = event => {
    setValue('photo', event.target.files[0])
    setPhoto(event.target.files[0])
  }

  return (
    <form onSubmit={onSubmit}>
      <Fields.Scanner
        error={errors.instrumentNumber}
        label="Instrument Number"
        setValue={value => setValue('instrumentNumber', value)}
        value={instrumentNumber}
      />
      <Fields.InstrumentTypeSelect
        error={errors.instrumentType}
        onChange={handleChange('instrumentType')}
        value={instrumentType}
        required
        data-testid="instrument-type"
      />
      <Fields.InstrumentSizeSelect
        error={errors.size}
        onChange={handleChange('size')}
        value={size}
        instrumentType={instrumentType}
        required
      />
      <Fields.LocationSelect
        error={errors.location}
        onChange={handleChange('location')}
        value={location}
        required
      />
      <FormControl fullWidth error={errors.assignedTo ? true : false}>
        <InputLabel htmlFor="assigned-to">Assigned To</InputLabel>
        <Input
          id="assigned-to"
          onChange={handleChange('assignedTo')}
          aria-describedby="assigned-to-error"
          type="text"
          value={assignedTo}
        />
        {errors.assignedTo && (
          <FormHelperText id="assigned-to-error">{errors.assignedTo}</FormHelperText>
        )}
      </FormControl>
      <FormControl fullWidth error={errors.maintenanceNotes ? true : false}>
        <InputLabel htmlFor="maintenance-notes">Maintenance Notes</InputLabel>
        <Input
          id="maintenance-notes"
          onChange={handleChange('maintenanceNotes')}
          aria-describedby="maintenance-notes-error"
          type="text"
          value={maintenanceNotes}
          multiline
        />
        {errors.maintenanceNotes && (
          <FormHelperText id="maintenance-notes-error">
            {errors.maintenanceNotes}
          </FormHelperText>
        )}
      </FormControl>
      <FormControl fullWidth error={errors.conditionNotes ? true : false}>
        <InputLabel htmlFor="condition-notes">Condition Notes</InputLabel>
        <Input
          id="condition-notes"
          onChange={handleChange('conditionNotes')}
          aria-describedby="condition-notes-error"
          type="text"
          value={conditionNotes}
          multiline
        />
        {errors.conditionNotes && (
          <FormHelperText id="condition-notes-error">
            {errors.conditionNotes}
          </FormHelperText>
        )}
      </FormControl>
      <FormControl fullWidth error={errors.condition ? true : false}>
        <InputLabel htmlFor="condition">Condition Rating</InputLabel>
        <Input
          type="number"
          id="condition"
          onChange={handleRating('condition')}
          aria-describedby="condition-error"
          value={condition}
        />
        {errors.condition && (
          <FormHelperText id="condition-error">{errors.condition}</FormHelperText>
        )}
      </FormControl>
      <FormControl fullWidth error={errors.quality ? true : false}>
        <InputLabel htmlFor="quality">Quality Rating</InputLabel>
        <Input
          type="number"
          id="quality"
          onChange={handleRating('quality')}
          aria-describedby="quality-error"
          value={quality}
        />
        {errors.quality && (
          <FormHelperText id="quality-error">{errors.quality}</FormHelperText>
        )}
      </FormControl>
      <FormGroup row>
        <FormControlLabel
          control={<Checkbox checked={rosin} id="rosin" onChange={handleCheck('rosin')} />}
          label="Rosin"
        />
        <FormControlLabel
          control={<Checkbox checked={bow} id="bow" onChange={handleCheck('bow')} />}
          label="Bow"
        />
        <FormControlLabel
          control={
            <Checkbox
              checked={shoulderRestEndpinRest}
              onChange={handleCheck('shoulderRestEndpinRest')}
              id="shoulder-rest-endpin-rest"
            />
          }
          label="Shoulder Rest/Rock Stop"
        />
        <FormControlLabel
          control={<Checkbox checked={readyToGo} id="ready-to-go" onChange={handleCheck('readyToGo')} />}
          label="Ready To Go"
        />
        <FormControlLabel
          control={<Checkbox checked={gifted} onChange={handleCheck('gifted')} />}
          label="Gifted To Student"
        />
      </FormGroup>
      {photoField && (
        <React.Fragment>
          {photo && <p>Photo: {photo.name}</p>}
          <input
            accept="image/*"
            id="upload-photo"
            type="file"
            className={classes.fileInput}
            onChange={handlePhoto}
          />
          <FormControl fullWidth error={errors.photo ? true : false}>
            <label htmlFor="upload-photo">
              <Button variant="contained" component="span" color="primary">
                Upload Photo
              </Button>
            </label>
            {errors.photo && (
              <FormHelperText id="photo-error">{errors.photo}</FormHelperText>
            )}
          </FormControl>
        </React.Fragment>
      )}
      <FormGroup row style={buttonsLeft ? { paddingTop: '0.5rem' } : {}}>
        {clearForm && (
          <Button onClick={clearForm} className={buttonsLeft ? '' : classes.lastButton}>
            Clear
          </Button>
        )}
        {onCancel && (
          <Button onClick={onCancel} className={buttonsLeft ? '' : classes.lastButton}>
            Cancel
          </Button>
        )}
        <Button type="submit" color="primary" disabled={!validateForm()}>
          Submit
        </Button>
      </FormGroup>
    </form>
  )
}

InstrumentForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  setValue: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired,
  setErrors: PropTypes.func.isRequired,
  instrumentNumber: PropTypes.string,
  instrumentType: PropTypes.string,
  size: PropTypes.string,
  location: PropTypes.string,
  assignedTo: PropTypes.string,
  maintenanceNotes: PropTypes.string,
  conditionNotes: PropTypes.string,
  condition: PropTypes.string,
  quality: PropTypes.string,
  rosin: PropTypes.bool,
  bow: PropTypes.bool,
  shoulderRestEndpinRest: PropTypes.bool,
  readyToGo: PropTypes.bool,
  gifted: PropTypes.bool,
  clearForm: PropTypes.func,
  validateForm: PropTypes.func.isRequired,
  onCancel: PropTypes.func,
  buttonsLeft: PropTypes.bool,
  photoField: PropTypes.bool,
}

export default InstrumentForm
