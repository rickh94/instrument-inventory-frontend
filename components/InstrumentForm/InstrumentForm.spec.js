import React from 'react'
import InstrumentForm from './InstrumentForm'
import { render, fireEvent } from '@testing-library/react'
import 'jest-dom/extend-expect'

const genericValues = {
  instrumentNumber: '1',
  instrumentType: 'violin',
  size: '4/4',
  location: 'office',
  assignedTo: 'test person',
  maintenanceNotes: 'test maintenance notes',
  conditionNotes: 'test condition notes',
  condition: '4',
  quality: '5',
  rosin: true,
  bow: false,
  shoulderRestEndpinRest: true,
  readyToGo: true,
  gifted: true,
}

describe('InstrumentForm', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <InstrumentForm
        {...genericValues}
        onSubmit={jest.fn()}
        setValue={jest.fn()}
        errors={{}}
        setErrors={jest.fn()}
        validateForm={jest.fn()}
        clearForm={jest.fn()}
        onCancel={jest.fn()}
        buttonsLeft={false}
        photoField={true}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it('passes props down to fields', () => {
    const { container } = render(
      <InstrumentForm
        {...genericValues}
        onSubmit={jest.fn()}
        setValue={jest.fn()}
        errors={{}}
        setErrors={jest.fn()}
        validateForm={jest.fn()}
        clearForm={jest.fn()}
        onCancel={jest.fn()}
        buttonsLeft={false}
        photoField={true}
      />
    )

    expect(container.querySelector('input#scanner-field')).toHaveAttribute(
      'value',
      genericValues.instrumentNumber
    )
    // expect(getByTestId('instrument-type')).toHaveAttribute('value', 'violin')
    expect(container.querySelector('input#assigned-to')).toHaveAttribute(
      'value',
      genericValues.assignedTo
    )
    expect(container.querySelector('textarea#maintenance-notes')).toHaveTextContent(
      genericValues.maintenanceNotes
    )
    expect(container.querySelector('textarea#condition-notes')).toHaveTextContent(
      genericValues.conditionNotes
    )
    expect(container.querySelector('input#condition')).toHaveAttribute(
      'value',
      genericValues.condition
    )
    expect(container.querySelector('input#quality')).toHaveAttribute(
      'value',
      genericValues.quality
    )
    expect(container.querySelector('#rosin')).toHaveAttribute('checked')
    expect(container.querySelector('#bow')).not.toHaveAttribute('checked')
    expect(container.querySelector('#shoulder-rest-endpin-rest')).toHaveAttribute(
      'checked'
    )
    expect(container.querySelector('#ready-to-go')).toHaveAttribute('checked')
  })

  it('calls the state update function on change', () => {
    const setValue = jest.fn()
    const { container } = render(
      <InstrumentForm
        {...genericValues}
        onSubmit={jest.fn()}
        setValue={setValue}
        errors={{}}
        setErrors={jest.fn()}
        validateForm={jest.fn()}
        clearForm={jest.fn()}
        onCancel={jest.fn()}
        buttonsLeft={false}
        photoField={true}
      />
    )
    fireEvent.change(container.querySelector('input#scanner-field'), {
      target: { value: 'newinstrumentnumber' },
    })
    expect(setValue).toHaveBeenCalledWith('instrumentNumber', 'newinstrumentnumber')

    fireEvent.change(container.querySelector('select#instrument-type'), {
      target: { value: 'viola' },
    })
    expect(setValue).toHaveBeenCalledWith('instrumentType', 'viola')

    fireEvent.change(container.querySelector('select#size'), {
      target: { value: '3/4' },
    })
    expect(setValue).toHaveBeenCalledWith('size', '3/4')

    fireEvent.change(container.querySelector('select#location'), {
      target: { value: 'Storage' },
    })
    expect(setValue).toHaveBeenCalledWith('location', 'Storage')

    fireEvent.change(container.querySelector('input#assigned-to'), {
      target: { value: 'test name' },
    })
    expect(setValue).toHaveBeenCalledWith('assignedTo', 'test name')

    fireEvent.change(container.querySelector('textarea#maintenance-notes'), {
      target: { value: 'new notes' },
    })
    expect(setValue).toHaveBeenCalledWith('maintenanceNotes', 'new notes')

    fireEvent.change(container.querySelector('textarea#condition-notes'), {
      target: { value: 'new notes' },
    })
    expect(setValue).toHaveBeenCalledWith('conditionNotes', 'new notes')

    fireEvent.change(container.querySelector('input#condition'), {
      target: { value: '2' },
    })
    expect(setValue).toHaveBeenCalledWith('condition', '2')

    fireEvent.change(container.querySelector('input#quality'), {
      target: { value: '4' },
    })
    expect(setValue).toHaveBeenCalledWith('quality', '4')

    // fireEvent.change(container.querySelector(''))
  })
})
