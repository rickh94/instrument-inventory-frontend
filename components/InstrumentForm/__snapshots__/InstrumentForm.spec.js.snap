// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentForm matches snapshot 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        data-testid="label"
        for="scanner-field"
      >
        Instrument Number
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        data-testid="input-field"
        required=""
      >
        <input
          aria-describedby="scanner-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          id="scanner-field"
          required=""
          type="text"
          value="1"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="scan-button"
            tabindex="0"
            title="Scan Barcode"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-barcode fa-w-16 "
                data-icon="barcode"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 448V64h18v384H0zm26.857-.273V64H36v383.727h-9.143zm27.143 0V64h8.857v383.727H54zm44.857 0V64h8.857v383.727h-8.857zm36 0V64h17.714v383.727h-17.714zm44.857 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm35.715 0V64h18v383.727h-18zm44.857 0V64h18v383.727h-18zm35.999 0V64h18.001v383.727h-18.001zm36.001 0V64h18.001v383.727h-18.001zm26.857 0V64h18v383.727h-18zm45.143 0V64h26.857v383.727h-26.857zm35.714 0V64h9.143v383.727H476zm18 .273V64h18v384h-18z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="MuiTypography-root MuiTypography-srOnly"
              >
                Scan Barcode
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="instrument-type"
      >
        Instrument Type
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        required=""
      >
        <div
          class="MuiNativeSelect-root"
        >
          <select
            aria-invalid="false"
            class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
            id="instrument-type"
            required=""
          >
            <option
              value=""
            />
            <option
              value="violin"
            >
              Violin
            </option>
            <option
              value="viola"
            >
              Viola
            </option>
            <option
              value="cello"
            >
              Cello
            </option>
            <option
              value="bass"
            >
              Bass
            </option>
            <option
              value="violin strug as viola"
            >
              Violin strung as viola
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiNativeSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="size"
      >
        Size
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        required=""
      >
        <div
          class="MuiNativeSelect-root"
        >
          <select
            aria-invalid="false"
            class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
            id="size"
            required=""
          >
            <option
              value=""
            />
            <option
              value="4/4"
            >
              4/4
            </option>
            <option
              value="3/4"
            >
              3/4
            </option>
            <option
              value="1/2"
            >
              1/2
            </option>
            <option
              value="1/4"
            >
              1/4
            </option>
            <option
              value="1/8"
            >
              1/8
            </option>
            <option
              value="1/10"
            >
              1/10
            </option>
            <option
              value="1/16"
            >
              1/16
            </option>
            <option
              value="1/32"
            >
              1/32
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiNativeSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="location"
      >
        Location
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        required=""
      >
        <div
          class="MuiNativeSelect-root"
        >
          <select
            aria-invalid="false"
            class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
            id="location"
            required=""
          >
            <option
              value=""
            />
            <option
              value="Grant Elementary School"
            >
              Grant Elementary School
            </option>
            <option
              value="Hedgepath Middle School"
            >
              Hedgepath Middle School
            </option>
            <option
              value="Wilson Elementary School"
            >
              Wilson Elementary School
            </option>
            <option
              value="Trenton High School"
            >
              Trenton High School
            </option>
            <option
              value="Columbus Elementary School"
            >
              Columbus Elementary School
            </option>
            <option
              value="office"
            >
              Office
            </option>
            <option
              value="Storage"
            >
              Storage
            </option>
            <option
              value="trade"
            >
              trade
            </option>
            <option
              value="maintenance"
            >
              Maintenance
            </option>
            <option
              value="transit"
            >
              Transit
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiNativeSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="assigned-to"
      >
        Assigned To
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-describedby="assigned-to-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="assigned-to"
          type="text"
          value="test person"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="maintenance-notes"
      >
        Maintenance Notes
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
      >
        <textarea
          aria-describedby="maintenance-notes-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          id="maintenance-notes"
          style="overflow: hidden;"
        >
          test maintenance notes
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="condition-notes"
      >
        Condition Notes
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
      >
        <textarea
          aria-describedby="condition-notes-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          id="condition-notes"
          style="overflow: hidden;"
        >
          test condition notes
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="condition"
      >
        Condition Rating
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-describedby="condition-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="condition"
          type="number"
          value="4"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
        for="quality"
      >
        Quality Rating
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-describedby="quality-error"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="quality"
          type="number"
          value="5"
        />
      </div>
    </div>
    <div
      class="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-139 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-140 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked=""
              class="PrivateSwitchBase-input-142"
              data-indeterminate="false"
              id="rosin"
              type="checkbox"
              value=""
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
        >
          Rosin
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-139 MuiCheckbox-root MuiCheckbox-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              class="PrivateSwitchBase-input-142"
              data-indeterminate="false"
              id="bow"
              type="checkbox"
              value=""
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
        >
          Bow
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-139 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-140 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked=""
              class="PrivateSwitchBase-input-142"
              data-indeterminate="false"
              id="shoulder-rest-endpin-rest"
              type="checkbox"
              value=""
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
        >
          Shoulder Rest/Rock Stop
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-139 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-140 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked=""
              class="PrivateSwitchBase-input-142"
              data-indeterminate="false"
              id="ready-to-go"
              type="checkbox"
              value=""
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
        >
          Ready To Go
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-139 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-140 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked=""
              class="PrivateSwitchBase-input-142"
              data-indeterminate="false"
              type="checkbox"
              value=""
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
        >
          Gifted To Student
        </span>
      </label>
    </div>
    <input
      accept="image/*"
      class="makeStyles-fileInput-2"
      id="upload-photo"
      type="file"
    />
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        for="upload-photo"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-containedPrimary MuiButton-contained"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Upload Photo
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </label>
    </div>
    <div
      class="MuiFormGroup-root MuiFormGroup-row"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-lastButton-1 MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Clear
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-lastButton-1 MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-text Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
</div>
`;
